### ISSUE TITLE
Creating a DX11 Swapchain with Immediate PresentMode Asserts Failure

### ISSUE THREAD

**Issue by cwfitzgerald** [gfx-rs/wgpu#1005]
**Description**

Creating a swapchain with immediate PresentationMode on DX11 crashes.

In general the DX11 swapchain creation code needs to be evaluated for compat across platforms.

**Expected vs observed behavior**

Expected: Beautiful pixels gracing my eyeballs
Observed:
```
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `-2005270527`,
 right: `0`', C:\Users\conno\.cargo\registry\src\github.com-1ecc6299db9ec823\gfx-backend-dx11-0.6.8\src\device.rs:764:13
```

**Platform**

Win10/Intel

**Comment by kvark** [comment]
Does DX11 backend report that this mode is supported?

**Comment by cwfitzgerald** [comment]
Fixed in gfx-backend-dx11 v0.6.12
